rapidtide/workflows/rapidtide_parser.py

```diff
@@ -21,176 +21,223 @@
 #
 #
 #
-from __future__ import print_function, division
+from __future__ import division, print_function
 
 import argparse
+import logging
 import sys
 
+import nibabel as nib
 import numpy as np
 
 import rapidtide.filter as tide_filt
 import rapidtide.io as tide_io
 import rapidtide.util as tide_util
-
-import nibabel as nib
-
 import rapidtide.workflows.parser_funcs as pf
 
-'''class indicatespecifiedAction(argparse.Action):
-    def __call__(self, parser, namespace, values, option_string=None):
-        setattr(namespace, self.dest, values)
-        setattr(namespace, self.dest+'_nondefault', True)
 
-def setifnotset(thedict, thekey, theval):
-    try:
-        test = thedict[thekey + '_nondefault']
-    except KeyError:
-        print('overriding ' + thekey)
-        thedict[thekey] = theval'''
+LGR = logging.getLogger(__name__)
 
 
 def _get_parser():
     """
     Argument parser for rapidtide
     """
-    parser = argparse.ArgumentParser(prog='rapidtide',
-                                     description='Perform a RIPTiDe time delay analysis on a dataset.',
-                                     usage='%(prog)s in_file outputname [options]')
+    parser = argparse.ArgumentParser(
+        prog="rapidtide",
+        description=("Perform a RIPTiDe time delay analysis on a dataset."),
+        usage="%(prog)s in_file outputname [options]",
+    )
 
     # Required arguments
     parser.add_argument(
-            'in_file',
-            type=lambda x: pf.is_valid_file(parser, x),
-            help='The input data file (BOLD fmri file or NIRS text file)')
+        "in_file",
+        type=lambda x: pf.is_valid_file(parser, x),
+        help="The input data file (BOLD fMRI file or NIRS text file)",
+    )
     parser.add_argument(
-            'outputname',
-            help='The root name for the output files.  For BIDS compliance, this can only contain valid BIDS entities from the source data.')
+        "outputname",
+        type=str,
+        help=(
+            "The root name for the output files.  "
+            "For BIDS compliance, this can only contain valid BIDS entities "
+            "from the source data."
+        ),
+    )
 
     # Analysis types
-    analysis_type = parser.add_argument_group('Analysis type').add_mutually_exclusive_group()
+    analysis_type = parser.add_argument_group(
+        title="Analysis type",
+        description=(
+            "Single arguments that change default values for many "
+            "arguments. "
+            "Any parameter set by an analysis type can be overridden "
+            "by setting that parameter explicitly. "
+            "Analysis types are mutually exclusive with one another, "
+            "but may be combined with 'Macros'."
```

Comment:
I hope this description is accurate. I wasn't entirely sure.

Reply:
Remove the last phrase, "but may be combined with 'Macros'" and I'd say it's good.